ecs-agent/api/resource/resource_attachment.go

```diff
@@ -27,6 +27,8 @@ import (
 
 type ResourceAttachment struct {
 	attachmentinfo.AttachmentInfo
+	// AttachmentType is the type of the resource attachment which can be "AmazonElasticBlockStorage" for EBS attach tasks.
+	AttachmentType string
```

Comment:
Q: should this also include `json:AttachmentType`?

Reply:
Although, it also doesn't seem like we're setting this field in this PR and I'm assuming there's going to be another PR that does so. Perhaps this field could be added to the [`AttachmentInfo`](https://github.com/aws/amazon-ecs-agent/blob/master/ecs-agent/api/attachmentinfo/attachment_info.go#L22) struct. I'm happy for either.

Reply:
> Q: should this also include json:AttachmentType?

Yes we can. It's optional. I'll add one.

In the ACS model, the attachmenttype and attachmentproperties are same level so I follow the same here
```
 type Attachment struct {
	_ struct{} `type:"structure"`

	AttachmentArn *string `locationName:"attachmentArn" type:"string"`

	AttachmentProperties []*AttachmentProperty `locationName:"attachmentProperties" type:"list"`

	AttachmentType *string `locationName:"attachmentType" type:"string"`
}
```